WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift

```diff
@@ -326,6 +340,18 @@ private extension CommentDetailViewController {
         view.addSubview(containerStackView)
         containerStackView.axis = .vertical
         containerStackView.addArrangedSubview(tableView)
+        if comment.allowsModeration(),
+            let moderationState = CommentModerationState(comment: comment) {
+            let commentModerationView = CommentModerationView(
+                viewModel: CommentModerationViewModel(
+                    state: moderationState,
+                    comment: comment,
+                    coordinator: CommentModerationCoordinator(commentDetailViewController: self)
+                )
+            )
+            let hostingController = UIHostingController(rootView: commentModerationView)
+            containerStackView.addArrangedSubview(hostingController.view)
+        }
```

Comment:
Hey @alpavanoglu, the moderation doesn't show up in my case even though `comment.allowsModeration` is `true` and comment status is `pending`. 

<img width=400 src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/73a68d79-d7c0-4642-b0b4-47f6ea0ab316" />

I did some digging and it turns out `CommentModerationState(comment: comment)` was `nil` because when a comment is pending, the `comment.status` value is `hold` and not `pending`.

After investigating the issue, I discovered that `CommentModerationState(comment: comment)` was returning `nil` because the `comment.status` for a pending comment is set to `hold`, not `pending`.

**Suggestion**: I think It would be more reliable to use the `CommentDetailViewController.commentStatus`, which is of type `CommentStatusType`. This property was used in the outdated moderation view.

Reply:
That type seem to have `draft` as an option. I don't think it is relevant in Notifications context. We would need to handle the `draft` case. I've updated the value to `hold` now. Lmk if you have other concerns regarding that.